### DISCUSSION TITLE
Protect user sign up from DDOS

### DISCUSSION THREAD

**Discussion by so-shiba** [supabase/supabase#27478] [Questions]
Firebase has limit on sign up, 100 accounts / hour /per id. 
How should I protect supabase user sign up by DDOS? The user sign up is linked to my own STMP. I want to prevent DOS on the STMP.

**Comment by GaryAustin1** [comment]
You can rate limit the email sending:
![image](https://github.com/supabase/supabase/assets/54564956/05f71003-7f59-436e-b184-12418d22b3cf)

Other limits: https://supabase.com/docs/guides/auth/rate-limits

**Comment by so-shiba** [comment]
Yes. I understand that. The documentation mentioned "Sum of combined
requests".

Is this per IP based limit? Or do all users share a single limit?

**Reply by GaryAustin1** [reply]
The email limit is max per hour from all potential users.  It throttles what will be sent to your SMTP provider.

**Comment by so-shiba** [comment]
Thank you. Please refer to my original question. How do I prevent malicious
attacker spam the sign up endpoint, causing denial of service of sign up
for normal users and increase bill of my STMP service?

**Reply by GaryAustin1** [reply]
Besides the rate limits shown, there is Captcha https://supabase.com/docs/guides/auth/auth-captcha?queryGroups=captcha-method&captcha-method=turnstile-1 .   

The only other thing I've seen is the CDN (cloudflare) helping block DDOS.  Mentioned here: https://supabase.com/security

I don't have more info for you as not something I've looked at.  Maybe another user or someone from SB will respond.
